internal/generate/generate.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generate
+
+// Generator is an implementation of the GeneratorSDK interface
+type Generator struct{}
+
+// GeneratorSDK is an interface that specifies the Generate methods
+// to generate and write various OperatorSDK types such as
+// package manifests
+type GeneratorSDK interface {
+	//GeneratePackageManifest(opts *PkgOptions) error
+	GeneratePackageManifest(operatorName, version, outputDir string, opts ...*PkgOptions) error
```

Comment:
Why we need `opts ...*PkgOptions` if in the code it is getting just the first value?

Reply:
...*PkgOptions is a variadic argument, which means it can be any number of PkgOptions. This is just some syntactic sugar so you can provide PkgOptions or not when you call the method - it's purely optional. It's never intended to be called with more than one. I can remove this if you think I'm trying to be too clever.